L1Trigger/Phase2L1Taus/plugins/HPSPFTauProducer.cc

```diff
@@ -0,0 +1,317 @@
+    if (passesSignalQualityCuts) {
+      selectedL1PFCandsSignalQualityCuts.push_back(l1PFCand);
+    }
+    if (passesSignalQualityCuts || passesIsolationQualityCuts) {
+      selectedL1PFCandsSignalOrIsolationQualityCuts.push_back(l1PFCand);
+    }
+  }
+
+  // sort PFCandidate collection by decreasing pT
+  std::sort(selectedL1PFCandsSignalQualityCuts.begin(), selectedL1PFCandsSignalQualityCuts.end(), isHigherPt_pfCandRef);
+  std::sort(selectedL1PFCandsSignalOrIsolationQualityCuts.begin(),
+            selectedL1PFCandsSignalOrIsolationQualityCuts.end(),
+            isHigherPt_pfCandRef);
+
+  if (debug_) {
+    std::cout << "AFTER selection (signalQualityCuts):" << std::endl;
+    for (auto l1PFCand : selectedL1PFCandsSignalQualityCuts) {
+      printPFCand(std::cout, *l1PFCand, primaryVertex_z);
+    }
+  }
+
+  l1t::HPSPFTauCollection l1PFTauCollectionUncleaned;
+
+  if (useChargedPFCandSeeds_) {
+    for (auto l1PFCand : selectedL1PFCandsSignalQualityCuts) {
+      if (l1PFCand->charge() != 0 && l1PFCand->pt() > minSeedChargedPFCandPt_ &&
+          std::fabs(l1PFCand->eta()) < maxSeedChargedPFCandEta_) {
+        bool isFromPrimaryVertex = false;
+        if (primaryVertex.get()) {
+          l1t::PFTrackRef l1PFTrack = l1PFCand->pfTrack();
+          double dz = std::fabs(l1PFTrack->vertex().z() - primaryVertex_z);
+          if (dz < maxSeedChargedPFCandDz_) {
+            isFromPrimaryVertex = true;
+          }
+        } else {
+          isFromPrimaryVertex = true;
+        }
+        if (isFromPrimaryVertex) {
+          tauBuilder_.reset();
+          tauBuilder_.setL1PFCandProductID(l1PFCands.id());
+          tauBuilder_.setVertex(primaryVertex);
+          tauBuilder_.setL1PFTauSeed(l1PFCand);
+          tauBuilder_.addL1PFCandidates(selectedL1PFCandsSignalOrIsolationQualityCuts);
+          tauBuilder_.buildL1PFTau();
+          l1t::HPSPFTau l1PFTau = tauBuilder_.getL1PFTau();
+          if (l1PFTau.pt() > 1.)
+            l1PFTauCollectionUncleaned.push_back(l1PFTau);
+        }
+      }
+    }
+  }
+
+  if (useJetSeeds_) {
+    edm::Handle<std::vector<reco::CaloJet>> l1Jets;
+    evt.getByToken(tokenL1Jets_, l1Jets);
+
+    size_t numL1Jets = l1Jets->size();
+    for (size_t idxL1Jet = 0; idxL1Jet < numL1Jets; ++idxL1Jet) {
+      reco::CaloJetRef l1Jet(l1Jets, idxL1Jet);
+      if (l1Jet->pt() > minSeedJetPt_ && std::fabs(l1Jet->eta()) < maxSeedJetEta_) {
+        tauBuilder_.reset();
+        tauBuilder_.setL1PFCandProductID(l1PFCands.id());
+        tauBuilder_.setVertex(primaryVertex);
+        //tauBuilder_.setL1PFTauSeed(l1Jet);
+        tauBuilder_.setL1PFTauSeed(l1Jet, selectedL1PFCandsSignalQualityCuts);
+        tauBuilder_.addL1PFCandidates(selectedL1PFCandsSignalOrIsolationQualityCuts);
+        tauBuilder_.buildL1PFTau();
+        l1t::HPSPFTau l1PFTau = tauBuilder_.getL1PFTau();
+        if (l1PFTau.pt() > 1.)
```

Comment:
its never a great idea to have a hardcode cut buried here. Must better to at least put it as a constant in the header.

Reply:
Done